src/sys/windows/udp.rs

```diff
@@ -57,23 +57,30 @@ impl UdpSocket {
     }
 
     pub fn send_to(&self, buf: &[u8], target: SocketAddr) -> io::Result<usize> {
-        try_io!(self, send_to, buf, target)
+        try_io!(
+            self.send_to(buf, target),
+            |bytes_written: &usize| *bytes_written <= buf.len()
+        )
     }
 
     pub fn recv_from(&self, buf: &mut [u8]) -> io::Result<(usize, SocketAddr)> {
-        try_io!(self, recv_from, buf)
+        try_io!(self.recv_from(buf), |t: &(usize, SocketAddr)| match t {
+            (bytes_read, _) => *bytes_read <= buf.len(),
```

Comment:
UDP doesn't work the same way TCP does. Calling `recv(_from)` will receive a single packet. Because of this`buf` is usually `1 << 16` bytes (~65k, the maximum size of a single packet) and most read packets will be smaller then that. Which means that it will always re-register, which emulating level triggers.

Reply:
In this case we need to always reregister for recv(_from) on successfull op.

Reply:
reverted this change, not needed.